sdk/src/main/java/com/hedera/hashgraph/sdk/TokenCreateTransaction.java

```diff
@@ -13,14 +13,19 @@
 public class TokenCreateTransaction extends Transaction<TokenCreateTransaction> {
     private final TokenCreateTransactionBody.Builder builder;
 
-    AccountId treasuryAccountId;
-    AccountId autoRenewAccountId;
+    @Nullable
+    CustomFeeList customFeeList = null;
+    @Nullable
+    AccountId treasuryAccountId = null;
+    @Nullable
+    AccountId autoRenewAccountId = null;
 
     public TokenCreateTransaction() {
         builder = TokenCreateTransactionBody.newBuilder();
 
         setAutoRenewPeriod(DEFAULT_AUTO_RENEW_PERIOD);
         setMaxTransactionFee(new Hbar(30));
+        setInitialSupply(0);
```

Comment:
We should not be setting the default initial supply. Let Hedera error here.

Reply:
I have a vague memory of having a good reason for doing this.  I'll talk to you about it tomorrow.

Reply:
From `TokenCreate.proto`:

`    uint64 initialSupply = 4; // Specifies the initial supply of tokens to be put in circulation. The initial supply is sent to the Treasury Account. The supply is in the lowest denomination possible. *In the case for NON_FUNGIBLE_UNIQUE Type the value must be 0*`

So this was my dirty solution.

Reply:
I guess what we should do then is set initial balance to 0 when someone `setTokenType(TokenType.NON_FUNGIBLE_UNIQUE)`, but not by default. Although, I'm pretty sure the default value for this field is 0 anyways so this is technically unneeded.